### PR TITLE
Set CSV header row from properties specified in the query

### LINKED ISSUE #3570: Datastore query download CSV header does not align with the requested data columns
## Description
User utilizes the datastore query download endpoint because they seek a dataset CSV containing specific columns only. The resulting CSV contains the data columns expected, but the header row contain every column header, even unwanted ones.

## Steps To Reproduce
1. Spin up a DKAN site with its demo data
2. Get the distribution's UUID of the Florida Bike Lanes dataset using `/api/1/metastore/schemas/dataset/items/cedcd327-4e5d-43f9-8eb1-c11850fa7c55?show_reference_ids`
3. POST on `/api/1/datastore/query/download` with the following payload `{"format": "csv","properties":["road_side","descr","shape_leng"],"resources":[{"id":"UUID","alias":"t"}]}`, replacing UUID with the above identifier
4. Resulting CSV contains:
```
objectid,roadway,road_side,lncd,descr,begin_post,end_post,shape_leng
R,DESIGNATED,463.2487
R,DESIGNATED,8267.4337
R,DESIGNATED,5430.1665
R,DESIGNATED,1825.1312
L,DESIGNATED,101.6744
R,DESIGNATED,185.7312
R,DESIGNATED,653.9896
L,DESIGNATED,80.7363
R,DESIGNATED,4964.3072
...
```

## Expected behavior
Header columns are correctly aligned with the data columns.
```
road_side,descr,shape_leng
R,DESIGNATED,463.2487
R,DESIGNATED,8267.4337
R,DESIGNATED,5430.1665
R,DESIGNATED,1825.1312
L,DESIGNATED,101.6744
R,DESIGNATED,185.7312
R,DESIGNATED,653.9896
L,DESIGNATED,80.7363
R,DESIGNATED,4964.3072
...
```

### PR DESCRIPTION
**Pull request by thierrydallacroce**
fixes #3570

## QA Steps

1. Spin up a DKAN site with its demo data
2. Get the distribution's UUID of the Florida Bike Lanes dataset using `/api/1/metastore/schemas/dataset/items/cedcd327-4e5d-43f9-8eb1-c11850fa7c55?show_reference_ids`
3. POST on `/api/1/datastore/query/download` with the following payload `{"format": "csv","properties":["road_side","descr","shape_leng"],"resources":[{"id":"UUID","alias":"t"}]}`, replacing UUID with the above identifier
1. Expected CSV contains:
```
road_side,descr,shape_leng
R,DESIGNATED,463.2487
R,DESIGNATED,8267.4337
R,DESIGNATED,5430.1665
R,DESIGNATED,1825.1312
L,DESIGNATED,101.6744
R,DESIGNATED,185.7312
R,DESIGNATED,653.9896
L,DESIGNATED,80.7363
R,DESIGNATED,4964.3072
...
```

### CODE REVIEW

File: modules/datastore/src/Controller/QueryController.php [hunk 1]
```diff
@@ -214,10 +214,17 @@ private function sendRows($handle, array $data) {
...
 
   /**
-   * Add the header row.
+   * Add the header row, from specified properties, if any, or the schema.
    */
   private function addHeaderRow(array &$data) {
-    $header_row = array_keys(reset($data['schema'])['fields']);
+
+    if (!empty($data['query']['properties'])) {
+      $header_row = $data['query']['properties'];
```
Comment (dafeder) [inline modules/datastore/src/Controller/QueryController.php:222]: @thierrydallacroce possible to provide test coverage for this condition?

**Review by dafeder — CHANGES_REQUESTED** [review]

File: modules/datastore/src/Controller/QueryController.php (hunk 1 above)
Comment (thierrydallacroce) [reply modules/datastore/src/Controller/QueryController.php:222]: Sure, trying

Comment (thierrydallacroce) [reply modules/datastore/src/Controller/QueryController.php:222]: Here you go, should be good.

Screen Shot 2021-07-20 at 5 28 13 PM

Comment (thierrydallacroce) [reply modules/datastore/src/Controller/QueryController.php:222]: Let me know if you have any questions or concerns.

**Pull request merged**